cpp/src/io/orc/orc_gpu.h

```diff
@@ -54,14 +56,15 @@ struct CompressedStreamInfo {
   }
   const uint8_t* compressed_data;  // [in] base ptr to compressed stream data
   uint8_t* uncompressed_data;  // [in] base ptr to uncompressed stream data or NULL if not known yet
-  size_t compressed_data_size;      // [in] compressed data size for this stream
-  gpu_inflate_input_s* decctl;      // [in] base ptr to decompression structure to be filled
-  gpu_inflate_status_s* decstatus;  // [in] results of decompression
-  gpu_inflate_input_s*
-    copyctl;  // [in] base ptr to copy structure to be filled for uncompressed blocks
+  size_t compressed_data_size;              // [in] compressed data size for this stream
+  device_span<uint8_t const>* dec_in_ctl;   // [in] input buffer to decompress
+  device_span<uint8_t>* dec_out_ctl;        // [in] output buffer to decompress into
+  decompress_status* decstatus;             // [in] results of decompression
```

Comment:
It's not really consistent to use a span here, since the other members are pointers. For example, `dec_in_ctl` should be a span of span, but the code requires it to be a pointer :vomiting_face: 
I changed it into a span anyway, every bit helps with this mess :)

Reply:
Oh, I didn't even see the `*` indicating that `dec_in_ctl` was a pointer. Otherwise I wouldn't have asked this. 🙃